### DISCUSSION TITLE
Expected latency?

### DISCUSSION THREAD

**Discussion by truher** [raspberrypi/picamera2#546] [General]
Hi picamera2 folk  I'm using a pi4 to do real time vision, looking for the lowest latency possible with the widest fov.  I'm using 2x2 binning full frame main to force the mode,  but encoding the lores which is half of that, to using pre_, callback Sandy mappedarray.  we see about 80ms from an event (e.g. a light turning on) to recognition by the pi.  Is 80ms a normal preprocessing latency?

**Comment by davidplowman** [comment]
Hi, this is often a difficult question to answer. 80ms sounds quite a lot but then at 30fps it can take some 30ms even to receive frame from the sensor, and the whole process is pipelined too. There might be some other factors, such as exactly how things are configured, and what other stuff is going on and in what threads. Do you have a very minimal example script to show what you're doing?
